sharktank/sharktank/types/tensors.py

```diff
@@ -752,10 +752,31 @@ class ShardedTensor(InferenceTensor):
     """
 
     def __init__(
-        self, *, shape: list[int], shard_dim: int | None, name: str = UnnamedTensorName
+        self,
+        *,
+        shape: list[int],
+        shard_dim: int | None,
+        name: str = UnnamedTensorName,
+        devices: Tuple[int],
+        pinned: bool,
     ):
         super().__init__(name=name, shape=shape)
         self.shard_dim = shard_dim
+        self._devices = devices
+        self._pinned = pinned
+
+    @abstractmethod
+    def clone(self, **kwargs) -> "ShardedTensor":
+        """Create a clone of this tensor with the given properties overridden."""
+        ...  # TODO: transform_globals, DeviceTensorTrait, etc. What else needs to be copied?
```

Comment:
@rsuderman 

Would this be better if I instead modified and made use of `._clone_with_globals`?

Reply:
This is probably good enough. We can always come back after and clean up if its not an issue.